src/mixins/object_geometry.mixin.js

```diff
@@ -183,67 +202,112 @@
     },
 
     /**
-     * Checks if object is contained within the canvas with current viewportTransform
-     * the check is done stopping at first point that appears on screen
+     * Checks if object is displayed within the bbox created by `tl` and `br`
+     * @param {fabric.Point} tl
+     * @param {fabric.Point} br
+     * @param {Boolean} [absolute] use coordinates without viewportTransform
      * @param {Boolean} [calculate] use coordinates of current position instead of .aCoords
      * @return {Boolean} true if object is fully or partially contained within canvas
      */
-    isOnScreen: function(calculate) {
-      if (!this.canvas) {
-        return false;
-      }
-      var pointTL = this.canvas.vptCoords.tl, pointBR = this.canvas.vptCoords.br;
-      var points = this.getCoords(true, calculate);
+    isOnBoundingBox: function (tl, br, absolute, calculate) {
```

Comment:
does this need to be a new public method? Do developers need it? i would prefer no.
Especially the absolute and calculate make it complicate to use as other methods we have.

Reply:
I can dump the absolute, it is useless. I just didn't know if I should break parity with the others.
I don't mind making it private but it's a good and useful method, not a private hack

Reply:
made private
should I dump the `absolute` arg?